### PR TITLE
Refactor(editor): move editor to codex-ui

### PR DESCRIPTION
**Pull request by kseniaSs**
- Move `downloadTools` from `useEditor` to separate service
- Move `useEditor.ts` to `codex-ui`

### CODE REVIEW

File: codex-ui/src/vue/editor/useEditor.ts [hunk 1]
```diff
@@ -36,20 +32,26 @@ interface UseEditorParams {
...
  */
-export function useEditor({ id, content, isReadOnly, onChange }: UseEditorParams): {
+export function useEditor({ id, content, isReadOnly, onChange, tools }: UseEditorParams): {
   isEmpty: Ref<boolean>;
   refresh: (data: OutputData) => void;
 } {
   /**
    * Editor instance
    */
+  // @ts-expect-error Ts config doesn't match with Editor
```
Comment (kseniaSs) [inline codex-ui/src/vue/editor/useEditor.ts:54]: We have different `tsconfig` from main. And if i change config in `tsconfig.json`, i have some errors in different ui component. So i have question, we change `tsconfig` and try to fix errors ?

File: codex-ui/package.json [hunk 2]
```diff
@@ -36,6 +36,20 @@
...
+    "@editorjs/embed": "^2.5.3",
+    "@editorjs/header": "^2.7.0",
+    "@editorjs/image": "^2.8.1",
+    "@editorjs/inline-code": "^1.4.0",
+    "@editorjs/link": "^2.5.0",
+    "@editorjs/list": "^1.8.0",
+    "@editorjs/marker": "^1.3.0",
+    "@editorjs/raw": "^2.4.0",
+    "@editorjs/table": "^2.2.2",
+    "@editorjs/warning": "^1.3.0",
```
Comment (TatianaFomina) [inline codex-ui/package.json:52]: why do we need to install the tools if we download them dynamically?

File: codex-ui/src/vue/editor/useEditor.ts (hunk 1 above)
Comment (TatianaFomina) [reply codex-ui/src/vue/editor/useEditor.ts:54]: what error does it display here?

File: codex-ui/src/vue/editor/useEditor.ts [hunk 3]
```diff
@@ -149,14 +124,14 @@ export function useEditor({ id, content, isReadOnly, onChange }: UseEditorParams
...
-      await editor.isReady;
+      await editor?.isReady;
     } catch (e) {
       console.error(e);
     }
   }
 
   /**
-   * Reinitializes editor instance with new data
+   * Reinitialized editor instance with new data
```
Comment (TatianaFomina) [inline codex-ui/src/vue/editor/useEditor.ts:134]:
```suggestion
   * Reinitializes editor instance with new data
```

File: codex-ui/src/vue/editor/useEditor.ts [hunk 4]
```diff
@@ -1,17 +1,13 @@
...
 import { onBeforeUnmount, onMounted, ref } from 'vue';
-import { useAppState } from './useAppState';
-import type EditorTool from '@/domain/entities/EditorTool';
-import { loadScript } from '@/infrastructure/utils/load-script';
 
 /**
  * Downloaded tools data structure
  */
-type DownloadedTools = Record<string, BlockTool>;
+export type DownloadedTools = Record<string, BlockTool>;
```
Comment (neSpecc) [inline codex-ui/src/vue/editor/useEditor.ts:10]: name  is unclear in this context.

File: codex-ui/src/vue/editor/Editor.vue [hunk 5]
```diff
@@ -28,6 +32,7 @@ const { isEmpty, refresh } = useEditor({
   id: 'editorjs',
   content: props.content,
   isReadOnly: props.readOnly,
+  tools: props.tools || {},
```
Comment (neSpecc) [inline codex-ui/src/vue/editor/Editor.vue:35]:
```suggestion
  tools: props.tools ?? {},
```

File: codex-ui/src/vue/editor/useEditor.ts (hunk 4 above)
Comment (neSpecc) [reply codex-ui/src/vue/editor/useEditor.ts:10]: its better to use Editor.js type instead (`EditorConfig["tools"]`)

**Review by TatianaFomina — APPROVED** [review]

File: codex-ui/package.json [hunk 6]
```diff
@@ -39,6 +39,7 @@
     "vue-tsc": "latest"
   },
   "dependencies": {
+    "@editorjs/editorjs": "^2.28.0",
```
Comment (neSpecc) [inline codex-ui/package.json:42]: lets use the latest version  (2.29.1)

File: package.json [hunk 7]
```diff
@@ -22,7 +22,7 @@
     "@editorjs/checklist": "^1.5.0",
     "@editorjs/code": "^2.8.0",
     "@editorjs/delimiter": "^1.3.0",
-    "@editorjs/editorjs": "^2.28.0",
+    "@editorjs/editorjs": "^2.29.1",
```
Comment (neSpecc) [inline package.json:25]: Seems like editor and its plugins are not used here.

File: codex-ui/package.json [hunk 8]
```diff
@@ -41,6 +41,8 @@
     "vue-tsc": "latest"
   },
   "dependencies": {
+    "@editorjs/editorjs": "^2.29.1",
+    "@editorjs/header": "^2.7.0",
```
Comment (neSpecc) [inline codex-ui/package.json:45]: why header is in dependencies?

**Review by neSpecc — CHANGES_REQUESTED** [review]

**Review by neSpecc — APPROVED** [review]

**Pull request merged**